packages/react/src/misc/adapters/useProfilePriceController.ts

```diff
@@ -0,0 +1,37 @@
+import { formatUnits } from '@ethersproject/units';
+import { UnspecifiedError } from '@lens-protocol/api-bindings';
+import { permissionlessCreator } from '@lens-protocol/blockchain-bindings';
+import {
+  Amount,
+  ChainType,
+  Matic,
+  PromiseResult,
+  assertError,
+  failure,
+  matic,
+  success,
+} from '@lens-protocol/shared-kernel';
+
+import { useSharedDependencies } from '../../shared';
+
+export function useProfilePriceController() {
+  const { config, providerFactory } = useSharedDependencies();
+
+  return async (): PromiseResult<Amount<Matic>, UnspecifiedError> => {
+    try {
+      const provider = await providerFactory.createProvider({ chainType: ChainType.POLYGON });
+      const contract = permissionlessCreator(
+        config.environment.contracts.permissionlessCreator,
+        provider,
+      );
+
+      const priceBN = await contract.getProfileWithHandleCreationPrice();
+      const result = formatUnits(priceBN, matic().decimals);
+
+      return success(Amount.matic(result));
```

Comment:
I have some experience recently with using Amount and different types of assets and I don't find it easy to work with, as you can see in the example above.